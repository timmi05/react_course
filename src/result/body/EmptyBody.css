.empty-body{
    font-size: 60px;
    height: 10em;
    display: flex;
    align-items: center;
    justify-content: center;
}