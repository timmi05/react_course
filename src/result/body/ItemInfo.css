.item-info-title-year{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    margin-bottom: 5px;
}

.item-info-title{
    font-size: 18px;
}

.item-info-release-year{
    font-size: 14px;
    border: 1px solid #ffffff;
    padding: 2px 10px;
    border-radius: 3px;
}

.item-info-tagline{
    font-size: 14px;
}