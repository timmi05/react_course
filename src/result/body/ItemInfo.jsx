import React from 'react';
import './item_info.css'

const ItemInfo = props => {
     const { title, tagline, release_date } = props;
     return (
          <div className='item-info'>
            <div className='item-info-title-year'>
                <div className='item-info-title'> { title } </div>
                <span className='item-info-release-year'>{ release_date }</span>
            </div>
            <div className='item-info-tagline'>
                { tagline }
             </div>
         </div>
     );
 };

export default ItemInfo