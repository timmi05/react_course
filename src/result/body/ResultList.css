.result-list{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    padding-inline-start: 0;
}