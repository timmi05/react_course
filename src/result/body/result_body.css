.result-body{
    display: flex;
    flex-direction: wrap;
    background: #232323;
}