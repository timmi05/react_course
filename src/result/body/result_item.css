.result-item{
    margin-bottom: 40px;
}

.result-item-cover{
    height: 500px;
    width: 350px;
    margin-bottom: 20px;
}