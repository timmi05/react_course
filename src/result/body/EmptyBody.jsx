import React from 'react';
import classnames from 'classnames';
import './EmptyBody.css'

export const EmptyBody = props => {
    const composedClassName = classnames(
        props.classesName,
        'empty-body'
      );
    return(
    <div className={ composedClassName }>No films found</div>
    )
};
