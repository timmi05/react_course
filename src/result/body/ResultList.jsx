import React from 'react';
import {ResultItem} from './ResultItem';
import './ResultList.css'

export const ResultList = props => {
    const {movies, openMovie, classesName} = props;
    return (
        <div className={classesName}>
            <ul className='result-list'>
                {movies.map(movie =>
                    < ResultItem key={movie.id}
                                 id={movie.id}
                                 posterPath={movie.poster_path}
                                 title={movie.title}
                                 tagline={movie.tagline}
                                 releaseDate={movie.release_date.substring(0, 4)}
                                 openMovie={openMovie}
                    />
                )}
            </ul>
        </div>
    )
};
