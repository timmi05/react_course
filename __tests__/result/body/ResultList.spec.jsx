import React from 'react';
import {shallow} from 'enzyme';
import {ResultList} from '../../../src/result/body/ResultList';

describe('ResultList rendering', () => {
    it('renders correctly', () => {
        const movies =
            [{
                id: 1,
                poster_path: 'posterPath',
                title: 'title',
                tagline: 'tagline',
                release_date: 'releaseDate'
            }];
        const component = shallow(<ResultList
            movies={movies}
        />);
        expect(component).toMatchSnapshot();
    });
});
