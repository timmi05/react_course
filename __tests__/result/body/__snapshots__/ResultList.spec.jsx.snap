// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultList rendering renders correctly 1`] = `
<div>
  <ul
    className="result-list"
  >
    <ResultItem
      id={1}
      key="1"
      posterPath="posterPath"
      releaseDate="rele"
      tagline="tagline"
      title="title"
    />
  </ul>
</div>
`;
